specification/app/resource-manager/Microsoft.App/preview/2022-01-01-preview/EasyAuthConfigs.json

```diff
@@ -0,0 +1,706 @@
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Easy Auth configuration resource specific properties",
+          "type": "object",
+          "properties": {
+            "enabled": {
+              "description": "<code>true</code> if the Authentication / Authorization feature is enabled for the current app; otherwise, <code>false</code>.",
+              "type": "boolean"
+            },
+            "globalValidation": {
+              "$ref": "#/definitions/GlobalValidation",
+              "description": "The configuration settings that determines the validation flow of users using ContainerApp Authentication/Authorization."
+            },
+            "identityProviders": {
+              "$ref": "#/definitions/IdentityProviders",
+              "description": "The configuration settings of each of the identity providers used to configure ContainerApp Authentication/Authorization."
+            },
+            "login": {
+              "$ref": "#/definitions/Login",
+              "description": "The configuration settings of the login flow of users using ContainerApp Authentication/Authorization."
+            },
+            "httpSettings": {
+              "$ref": "#/definitions/HttpSettings",
+              "description": "The configuration settings of the HTTP requests for authentication and authorization requests made against ContainerApp Authentication/Authorization."
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "GlobalValidation": {
+      "description": "The configuration settings that determines the validation flow of users using ContainerApp Authentication/Authorization.",
+      "type": "object",
+      "properties": {
+        "unauthenticatedClientAction": {
+          "description": "The action to take when an unauthenticated client attempts to access the app.",
+          "enum": [
+            "RedirectToLoginPage",
+            "AllowAnonymous",
+            "Return401",
+            "Return403"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "UnauthenticatedClientAction",
+            "modelAsString": true
+          }
+        },
+        "redirectToProvider": {
+          "description": "The default authentication provider to use when multiple providers are configured.\nThis setting is only needed if multiple providers are configured and the unauthenticated client\naction is set to \"RedirectToLoginPage\".",
+          "type": "string"
+        }
+      }
+    },
+    "Login": {
+      "description": "The configuration settings of the login flow of users using ContainerApp Authentication/Authorization.",
+      "type": "object",
+      "properties": {
+        "routes": {
```

Comment:
>routes [](http://example.com/codeflow?start=9&length=6)

Is this a collection? If so, please use an array rather than encoded string. If not, the property name should be singular.

Reply:
Good suggestion, will update to use singular.